Cargo.toml

```diff
@@ -37,6 +36,70 @@ rust.unexpected_cfgs = { level = "warn", check-cfg = [
 # (And cargo doesn't let us have platform specific lints here)
 rust.unsafe_code = "deny"
 
+rust.keyword_idents_2024 = "forbid"
+rust.non_ascii_idents = "forbid"
+rust.unsafe_op_in_unsafe_fn = "forbid"
+rust.non_local_definitions = "forbid"
+
+rust.unused_lifetimes = "warn"
+rust.unit_bindings = "warn"
+rust.unused_import_braces = "warn"
+rust.trivial_numeric_casts = "warn"
+rust.unused_macro_rules = "warn"
+rust.variant_size_differences = "warn"
+
+clippy.allow_attributes_without_reason = "warn"
```

Comment:
Hmm this, pollutes my diagnostics in my editor, without being able to action about it, since this does seem to be experimental, and thus CI fails when adding a reason (as seen [here](https://github.com/linebender/xilem/actions/runs/9999775115/job/27641321539#step:7:398)) , should we disable that for now, and enable it, when it's stable, maybe with directly fixing all of this, so that it doesn't contribute to OCD^^?

Reply:
We can comment this out.

However, the behaviour you're seeing is a bug in clippy. This should be changed in clippy to properly handle the MSRV for where this feature can be used.

Reply:
Hmm that's interesting (just updated nightly, and still the issue).
I don't use rust via rustup, but via [rust-overlay](https://github.com/oxalica/rust-overlay) maybe that's an issue, as there's no rustup-binary on my system. I mean it's not critical to me, I can just comment it out locally.